### PR TITLE
Typescript migration

### PR DESCRIPTION
**Pull request by ayushsatyam146**
Completed typescript migration of all components, utils, hooks, and pages of the project.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -15,5 +15,13 @@
   "plugins": ["react", "@typescript-eslint"],
   "rules": {
     "react/prop-types": "off"
-  }
+  },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "rules": {
+        "no-undef": "off"
```
Comment (Abhirup-99) [inline .eslintrc.json]: why was this rule added?

File: components/AboutBlock.tsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React from 'react'
 import Image from 'next/image'
+import { AboutBlockProps } from '../interface'
 
-export const AboutBlock = ({
+export const AboutBlock: React.FC<AboutBlockProps> = ({
```
Comment (Abhirup-99) [inline components/AboutBlock.tsx:5]: Can we have interfaces for all the components that take props?

File: components/CookiesBanner.tsx [hunk 3]
```diff
@@ -5,10 +5,10 @@ import Button from './Button'
 import Cookies from 'js-cookie'
 
 const CookiesBanner = () => {
-  const [cookiesBanner, setCookiesBanner] = useState(false)
+  const [cookiesBanner, setCookiesBanner] = useState<boolean>(false)
 
   /* Toggle customizable functionality */
-  const cookiesHandler = (callback) => {
+  const cookiesHandler = (callback: void) => {
```
Comment (Abhirup-99) [inline components/CookiesBanner.tsx]: this should be callback: `() => void`

File: components/CustomizeIconsPanel.tsx [hunk 4]
```diff
@@ -19,17 +19,17 @@ const sendData = async (params) => {
   return response.data
 }
 
-const downloadFont = (props) => {
+const downloadFont = (props: { timestamp: AxiosResponse | undefined }) => {
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: better written as `timestamp?: AxiosResponse`.

File: components/CustomizeIconsPanel.tsx [hunk 5]
```diff
@@ -19,17 +19,17 @@ const sendData = async (params) => {
...
   return window.open(downloadEndPoints, '_blank')
 }
 
-const CustomizeIconsPanel = (props) => {
+const CustomizeIconsPanel: React.FC<CustomizeIconsPanelProps> = (props) => {
   const { state, dispatch } = useContext(AppContext)
   const [iconEditor, setIconEditor] = useState(false)
 
-  const iconEditorToggle = (e) => {
+  const iconEditorToggle = (e: MouseEvent) => {
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: This should be a React Mouse Event. This is a React Synthetic Event not a DOM Event.

File: components/CustomizeIconsPanel.tsx [hunk 6]
```diff
@@ -38,13 +38,13 @@ const CustomizeIconsPanel = (props) => {
   const { selectAll, deselectAll } = props
 
   const [modal, setModal] = useState(false)
-  const [serverResponse, setServerResponse] = useState(null)
+  const [serverResponse, setServerResponse] = useState()
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Can we add an interface for what the server response is? Also, why was `null` removed?

File: components/CustomizeIconsPanel.tsx [hunk 7]
```diff
@@ -38,13 +38,13 @@ const CustomizeIconsPanel = (props) => {
...
 
   const modalToggle = () => {
     setModal(!modal)
   }
 
-  const generateFont = (e) => {
+  const generateFont = (e: MouseEvent) => {
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Add react synthetic event instead of the DOM event type.

File: components/CustomizeIconsPanel.tsx [hunk 8]
```diff
@@ -73,10 +73,10 @@ const CustomizeIconsPanel = (props) => {
         <div className='generate-div'>
           <span>{state.multipleIcons.length} icons selected</span>
           <span>Export as: </span>
-          <Button type='submit' onClick={generateFont}>
+          <Button type='submit' onClick={generateFont as () => void}>
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Don't typecast.

File: components/CustomizeIconsPanel.tsx [hunk 9]
```diff
@@ -73,10 +73,10 @@ const CustomizeIconsPanel = (props) => {
...
             Font
           </Button>
-          <Button type='button' onClick={iconEditorToggle}>
+          <Button type='button' onClick={iconEditorToggle as () => void}>
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Same here.

File: components/CustomizeIconsPanel.tsx [hunk 10]
```diff
@@ -93,15 +93,15 @@ const CustomizeIconsPanel = (props) => {
...
+              onChange={(event) => search(event.target.files![0], dispatch)}
             />
           </div>
         </div>
       </div>
       {iconEditor ? (
         <IconEditor
           isActive={iconEditor}
-          show={iconEditorToggle}
+          show={iconEditorToggle as () => void}
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Same here

File: components/CustomizeIconsPanel.tsx [hunk 11]
```diff
@@ -125,11 +125,11 @@ const CustomizeIconsPanel = (props) => {
   )
 }
 
-const search = (file, dispatch) => {
+const search = (file: File, dispatch: React.Dispatch<any>) => {
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Can we skip `any` here?

File: components/CustomizeIconsPanel.tsx [hunk 12]
```diff
@@ -125,11 +125,11 @@ const CustomizeIconsPanel = (props) => {
...
   const fileReader = new window.FileReader()
 
   fileReader.onload = function (fileData) {
-    const iconsArray = JSON.parse(fileData.target.result)
+    const iconsArray = JSON.parse(fileData.target!.result as string)
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Why was it typecasted?

File: components/Footer.tsx [hunk 13]
```diff
@@ -2,27 +2,34 @@ import React from 'react'
...
-      <div className='container flex flex-wrap'>
+    links && (
+      <footer>
         <div className='container flex flex-wrap'>
-          {links.map((ele, i) => {
-            return <FooterBlock {...ele} key={i} />
-          })}
+          <div className='container flex flex-wrap'>
+            {links.map((ele, i) => {
+              return <FooterBlock {...ele} key={i} />
```
Comment (Abhirup-99) [inline components/Footer.tsx]: Can we use `uuid` here instead of `i` for the key? Using the array position as the key is not considered a good practice.

File: components/Footer.tsx [hunk 14]
```diff
@@ -2,27 +2,34 @@ import React from 'react'
...
+            })}
+          </div>
+          <Link href={links[0].vercel!.href}>
+            <a className='banner' target={links[0].vercel!.target}>
+              <Image
+                className=''
+                src={links[0].vercel!.img}
```
Comment (Abhirup-99) [inline components/Footer.tsx]: From ts docs: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-

Can we use `links[0].vercel?.target || ""`?

File: components/HowToPanel.tsx [hunk 15]
```diff
@@ -4,23 +4,32 @@ import { IconSetContext } from '../utils/IconSetContext'
...
+  close,
+  iconName,
+  type,
+  iconTags
+}) => {
   const { iconDispatch } = useContext(IconSetContext)
-  const ref = useRef()
+  const ref = useRef<HTMLDivElement>(null)
   const [iconEditor, setIconEditor] = useState(false)
   const [iconType, setIconType] = useState('static')
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: We can't use all strings here, right? A better alternative:
`type IconType = "static" | "dynamic"`

File: components/HowToPanel.tsx [hunk 16]
```diff
@@ -4,23 +4,32 @@ import { IconSetContext } from '../utils/IconSetContext'
...
-  const iconEditorToggle = (type) => {
+
+  const iconEditorToggle = (type: 'animated' | 'static') => {
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: Can be reused here.

File: components/HowToPanel.tsx [hunk 17]
```diff
@@ -4,23 +4,32 @@ import { IconSetContext } from '../utils/IconSetContext'
...
 
   const { state, dispatch } = useContext(AppContext)
-  function useOnClickOrEsc(ref, handler) {
+  function useOnClickOrEsc(
+    ref: React.RefObject<HTMLDivElement>,
+    handler: (e: KeyboardEvent) => void
+  ) {
     useEffect(() => {
-      const listenerKeydown = (event) => {
+      const listenerKeydown = (event: KeyboardEvent) => {
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx:37]: Use React Event.

File: components/HowToPanel.tsx [hunk 18]
```diff
@@ -39,17 +48,18 @@ const HowToPanel = (props) => {
   if (typeof window !== 'undefined') {
     queryString = window.location.search
   }
-  const urlParams = new URLSearchParams(queryString)
-  const urlIconName = urlParams.get('iconName')
-  const urlTagName = urlParams.get('tagName')
+  const urlParams: URLSearchParams = new URLSearchParams(queryString)
+  const urlIconName: string | null = urlParams.get('iconName')
+  const urlTagName: string | null = urlParams.get('tagName')
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: better alternative: `?: string`

File: components/HowToPanel.tsx [hunk 19]
```diff
@@ -39,17 +48,18 @@ const HowToPanel = (props) => {
...
-  let setSearchWithUrlParam = urlIconName && !iconName ? urlIconName : ''
+  let setSearchWithUrlParam: string | null =
+    urlIconName && !iconName ? urlIconName : ''
 
   if (setSearchWithUrlParam === '') {
     setSearchWithUrlParam = urlTagName
   }
 
-  const selectTag = (urlTagName, callback) => {
+  const selectTag = (urlTagName: string, callback: void) => {
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: For callback: `() => void`

File: components/HowToPanel.tsx [hunk 20]
```diff
@@ -88,13 +98,16 @@ const HowToPanel = (props) => {
                 <input
                   id='copy-code'
                   className='input-group-element'
-                  readOnly='readOnly'
+                  readOnly={true}
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: you don't need `={true}`, only `readOnly` works the same. Also, it's cleaner to read.

File: components/HowToPanel.tsx [hunk 21]
```diff
@@ -117,7 +130,7 @@ const HowToPanel = (props) => {
...
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: Same here.

File: components/HowToPanel.tsx [hunk 22]
```diff
@@ -88,13 +98,16 @@ const HowToPanel = (props) => {
...
                   value={`<Image src='${iconName}.svg'/>`}
                 />
                 <Button
                   type='button'
                   onClick={() => {
-                    document.getElementById('copy-code').select()
+                    const element = document.getElementById(
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: This snippet seems to be duplicated in multiple places. Can we make this a function?

File: components/HowToPanel.tsx [hunk 23]
```diff
@@ -127,7 +140,10 @@ const HowToPanel = (props) => {
...
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: Duplicated here.

File: components/HowToPanel.tsx [hunk 24]
```diff
@@ -171,18 +187,18 @@ const HowToPanel = (props) => {
         {iconEditor ? (
           <IconEditor
             isActive={iconEditor}
-            show={iconEditorToggle}
+            show={iconEditorToggle as () => void}
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: Don't typecast.

File: components/HowToPanel.tsx [hunk 25]
```diff
@@ -171,18 +187,18 @@ const HowToPanel = (props) => {
...
           />
         ) : (
-          ''
+          <></>
         )}
       </div>
     </div>
   ) : (
-    ''
+    <></>
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: `null` is preferred over the `Fragment` syntax.

File: components/IconEditor.tsx [hunk 26]
```diff
@@ -53,27 +58,32 @@ const IconEditor = (props) => {
...
   // }
   const changeExportSize = () => {
-    setExportSize(document.getElementsByClassName('export-size')[0].value)
+    setExportSize(
+      (document.getElementsByClassName('export-size')[0] as HTMLInputElement)
+        .value
+    )
   }
-  const flipIconHorizontal = (e) => {
+  const flipIconHorizontal = (e: MouseEvent) => {
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Use React Events.

File: components/IconEditor.tsx [hunk 27]
```diff
@@ -53,27 +58,32 @@ const IconEditor = (props) => {
...
     e.preventDefault()
     setHorizontalFlip(!horizontalFlip)
   }
 
-  const flipIconVertical = (e) => {
+  const flipIconVertical = (e: MouseEvent) => {
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Use React Events.

File: components/IconEditor.tsx [hunk 28]
```diff
@@ -133,13 +143,13 @@ const IconEditor = (props) => {
     verticalFlip
   ])
 
-  const postDataToApi = async (params) => {
+  const postDataToApi = async (params: { url: string; payload: {} }) => {
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Can we type the payload part?

File: components/IconEditor.tsx [hunk 29]
```diff
@@ -150,7 +160,7 @@ const IconEditor = (props) => {
     }
     return window.open(downloadEndPoints, '_blank')
   }
-  const generateCustomizedIcon = (e) => {
+  const generateCustomizedIcon = (e: MouseEvent) => {
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Use React Events.

File: components/IconEditor.tsx [hunk 30]
```diff
@@ -198,15 +211,19 @@ const IconEditor = (props) => {
                     <input
                       id='copy-svg'
                       className='input-group-element'
-                      readOnly='readOnly'
+                      readOnly={true}
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Fix `readOnly`.

File: components/IconEditor.tsx [hunk 31]
```diff
@@ -198,15 +211,19 @@ const IconEditor = (props) => {
...
                       type='button'
                       disabled={!svgCode.length}
                       onClick={() => {
-                        document.getElementById('copy-svg').select()
+                        ;(
+                          document.getElementById(
+                            'copy-svg'
+                          ) as HTMLInputElement
+                        ).select()
                         document.execCommand('copy')
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Duplicated code.

File: components/IconEditor.tsx [hunk 32]
```diff
@@ -275,10 +292,10 @@ const IconEditor = (props) => {
               <div>
                 <p>Flip</p>
                 <div>
-                  <button onClick={flipIconHorizontal}>
+                  <button onClick={flipIconHorizontal as () => void}>
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Don't typecast.

File: components/IconEditor.tsx [hunk 33]
```diff
@@ -275,10 +292,10 @@ const IconEditor = (props) => {
...
                     <i className='eos-icons'>flip</i>
                   </button>
-                  <button onClick={flipIconVertical}>
+                  <button onClick={flipIconVertical as () => void}>
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Don't typecast.

File: components/IconEditor.tsx [hunk 34]
```diff
@@ -324,7 +341,10 @@ const IconEditor = (props) => {
               ''
             )}
             <div className='export-btn'>
-              <Button type='button' onClick={generateCustomizedIcon}>
+              <Button
+                type='button'
+                onClick={generateCustomizedIcon as () => void}
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: Don't typecast.

File: components/IconEditor.tsx [hunk 35]
```diff
@@ -344,7 +364,7 @@ const IconEditor = (props) => {
       </div>
     </div>
   ) : (
-    ''
+    <></>
```
Comment (Abhirup-99) [inline components/IconEditor.tsx]: use `null` over `React.Fragment` syntax.

File: components/IconsSet.tsx [hunk 36]
```diff
@@ -55,7 +56,7 @@ const IconsSet = (props) => {
               }`}
               onClick={() => {
                 if (iconState.searchValue.length > 0) {
-                  searchRef.current.value = ''
+                  searchRef.current!.value = ''
```
Comment (Abhirup-99) [inline components/IconsSet.tsx]: Don't use `!.` syntax.

File: components/IconsSet.tsx [hunk 37]
```diff
@@ -213,47 +206,49 @@ const IconsSet = (props) => {
...
-                            selection: icon.name
-                          })
-                        }}
-                        onDoubleClickAction={() => {
-                          iconDispatch({ type: 'TOGGLE_ICONEDITOR' })
-                        }}
-                      />
-                    ))}
+            {state.iconsCategory.map(
+              (categoryObject: Category, index: number) => {
```
Comment (Abhirup-99) [inline components/IconsSet.tsx:211]: Same point, use `uuid`.

File: components/Tabs.tsx [hunk 38]
```diff
@@ -30,7 +28,7 @@ const Tabs = (props) => {
   }, [currentTab])
 
   useEffect(() => {
-    setPosition(document.querySelector('.page-header').clientHeight + 54)
+    setPosition(document.querySelector('.page-header')!.clientHeight + 54)
```
Comment (Abhirup-99) [inline components/Tabs.tsx]: same.

File: components/Tabs.tsx [hunk 39]
```diff
@@ -113,8 +99,8 @@ const Tabs = (props) => {
       </ul>
 
       <div className='tab-content'>
-        {children.map((child) => {
-          if (child.props.label !== activeTab) return undefined
+        {children!.map((child) => {
+          if (child.props.htmlFor !== activeTab) return undefined
```
Comment (Abhirup-99) [inline components/Tabs.tsx]: return `null`. returning `undefined` is not the preferred way.

File: components/TeamBlock.tsx [hunk 40]
```diff
@@ -20,19 +30,30 @@ export const TeamBlock = ({ image, role, name, description, social }) => {
...
+              {social?.map(
+                (
+                  ele: {
+                    title: string
+                    link: string
+                  },
+                  i: number
+                ) => {
+                  return (
+                    <li key={i}>
```
Comment (Abhirup-99) [inline components/TeamBlock.tsx]: use `uuid`.

File: tsconfig.json [hunk 41]
```diff
@@ -13,7 +13,7 @@
     "moduleResolution": "node",
     "noEmit": true,
     "noFallthroughCasesInSwitch": true,
-    "noImplicitReturns": true,
+    "noImplicitReturns": false,
```
Comment (Abhirup-99) [inline tsconfig.json:16]: Why was this changed?

File: utils/EosIcons.store.ts [hunk 42]
```diff
@@ -26,46 +32,48 @@ const iconsCategory = categories.map((category) => {
...
-  selectAllTagsIcons: (tagName, iconsCategory) => {
+  selectAllTagsIcons: (tagName: string, iconsCategory: Category[]) => {
     if (tagName === 'all') return iconsCategory
-
     tagName = tagName.toLowerCase()
     return iconsCategory.map((ele) => {
       return {
         category: ele.category,
-        icons: ele.icons.filter((ele) => ele.tags.includes(tagName))
+        icons: ele.icons.filter((ele) => ele!.tags.includes(tagName))
```
Comment (Abhirup-99) [inline utils/EosIcons.store.ts]: Same.

File: utils/EosIcons.store.ts [hunk 43]
```diff
@@ -129,19 +141,19 @@ export const eosIconsState = {
         category: ele.category,
         icons: ele.icons.filter(
           (ele) =>
-            searchMutliple(ele.name, searchArray) ||
-            searchMutliple(ele.tags, searchArray)
+            searchMutliple([ele!.name], searchArray) ||
```
Comment (Abhirup-99) [inline utils/EosIcons.store.ts]: Same.

File: .eslintrc.json (hunk 1 above)
Comment (ayushsatyam146) [reply .eslintrc.json]: I added this rule to avoid getting es-lint errors for JSX import in the `interface/index.ts` file. We can either add it to globals or disable the `no-undef` rule because TypeScript already has already its own checks. 
I can add `globals` instead of this rule if wanted.

File: components/AboutBlock.tsx (hunk 2 above)
Comment (ayushsatyam146) [reply components/AboutBlock.tsx:5]: We have interfaces for all such components in `interface/index.ts`

File: .eslintrc.json (hunk 1 above)
Comment (Abhirup-99) [reply .eslintrc.json]: Which part was throwing the error?

File: utils/EosIcons.store.ts (hunk 43 above)
Comment (ayushsatyam146) [reply utils/EosIcons.store.ts]: I was getting this error **Object is possibly 'null'**. I just applied null check operator to ensure object is not undefined or null.

Comment (Abhirup-99) [reply utils/EosIcons.store.ts]: use `ele?.name || ""`

File: .eslintrc.json (hunk 1 above)
Comment (Abhirup-99) [reply .eslintrc.json]: I am assuming it's for React. Why are we not importing React explicitly then?

File: tsconfig.json (hunk 41 above)
Comment (ayushsatyam146) [reply tsconfig.json:16]: In `./IconSet.store'` file there are some functions in which there are no returns for certain control flows. These functions are consumed by the reducer so I can't put some dummy value for every flow, hence I changed this rule.

File: components/Tabs.tsx (hunk 39 above)
Comment (ayushsatyam146) [reply components/Tabs.tsx]: Thanks, I will do that

File: .eslintrc.json (hunk 1 above)
Comment (ayushsatyam146) [reply .eslintrc.json]: An error was for importing JSX.Element in this file. I'll add globals and improve it.

Comment (Abhirup-99) [reply .eslintrc.json]: make it, `interface/index.tsx`

Comment (Abhirup-99) [reply .eslintrc.json]: Use React.ReactNode instead of JSX.Element, and you can drop the globals :). I kept it `index.ts` only.

File: components/HowToPanel.tsx [hunk 44]
```diff
@@ -44,13 +46,9 @@ const HowToPanel: React.FC<HowToPanelProps> = ({
...
-  let queryString
-  if (typeof window !== 'undefined') {
-    queryString = window.location.search
-  }
-  const urlParams: URLSearchParams = new URLSearchParams(queryString)
-  const urlIconName: string | null = urlParams.get('iconName')
-  const urlTagName: string | null = urlParams.get('tagName')
+
+  const urlTagName = router.query.tagName as string
+  const urlIconName = router.query.iconName as string
```
Comment (Abhirup-99) [inline components/HowToPanel.tsx]: why was the snippet updated? How can we sure that query would always contain the `tagName`, and the `iconName`

Comment (Abhirup-99) [reply components/HowToPanel.tsx]: My comment was to accommodate the types part. The suggestion was to make it `const urlTagName : ?string = urlParams.get('iconName')`.

Comment (ayushsatyam146) [reply components/HowToPanel.tsx]: When you pointed at this section of code, I noticed that it was using the old way of fetching URL parameters. Using the router to fetch URL params is the preferred way in Next.js. I updated it to use router instead. 
URL doesn't contain `tagName` and `iconName` always, in that case, the values are undefined. We are putting checks in place to ensure that these values are defined where we need them.
I have typecasted the value to string because by default the value can have string[] and undefined types as well but in our use case it's always a string.

File: utils/IconSet.store.ts [hunk 45]
```diff
@@ -370,8 +371,8 @@ export const iconSetHelper = {
           ) {
             const icon = eosIconsState.iconsCategory[i].icons[j]
             if (
-              icon!.name.includes(words![k].toLowerCase()) ||
-              icon!.tags.includes(words![k].toLowerCase())
+              icon?.name.includes(words![k].toLowerCase()) ||
```
Comment (Abhirup-99) [inline utils/IconSet.store.ts]: `words!` is better as `words?.[i]?.toLowerCase()`

Comment (ayushsatyam146) [reply utils/IconSet.store.ts]: Hi, I have formatted the code to address the issue mentioned above. Please review.

File: .eslintrc.json [hunk 46]
```diff
@@ -15,5 +15,8 @@
   "plugins": ["react", "@typescript-eslint"],
   "rules": {
     "react/prop-types": "off"
+  },
+  "globals": {
+    "JSX": true
```
Comment (Abhirup-99) [inline .eslintrc.json]: Can we get this removed?

File: components/IconsSet.tsx (hunk 36 above)
Comment (Abhirup-99) [reply components/IconsSet.tsx]: Not fixed.

File: .eslintrc.json (hunk 46 above)
Comment (ayushsatyam146) [reply .eslintrc.json]: I have removed this and also fixed the other bug in `IconSet.tsx` file. Please check.

File: components/HowToPanel.tsx (hunk 17 above)
Comment (Abhirup-99) [reply components/HowToPanel.tsx:37]: Not updated?

File: components/CustomizeIconsPanel.tsx [hunk 47]
```diff
@@ -93,7 +94,7 @@ const CustomizeIconsPanel = (props) => {
               accept='application/JSON'
               hidden
               name='file'
-              onChange={(event) => search(event.target.files[0], dispatch)}
+              onChange={(event) => search(event.target.files![0], dispatch)}
```
Comment (Abhirup-99) [inline components/CustomizeIconsPanel.tsx]: Not updated?

File: interface/index.ts [hunk 48]
```diff
@@ -0,0 +1,214 @@
...
+  img?: any
+  title: string
+  links: link[]
+}
+
+export interface HowToPanelProps {
+  show: boolean
+  close: () => void
+  iconName: string
+  type: 'static' | 'animated'
```
Comment (Abhirup-99) [inline interface/index.ts]: use `IconType`

File: interface/index.ts [hunk 49]
```diff
@@ -0,0 +1,214 @@
...
+  iconTags: string[]
+}
+
+export interface IconProps {
+  name: string
+  size: number
+  onClickAction: () => void
+  onDoubleClickAction: () => void
+  type: 'static' | 'animated'
```
Comment (Abhirup-99) [inline interface/index.ts]: same!

File: components/CustomizeIconsPanel.tsx (hunk 47 above)
Comment (ayushsatyam146) [reply components/CustomizeIconsPanel.tsx]: Resolved. Please check.

**Review by Abhirup-99 — APPROVED** [review]
